ssl/t1_lib.c

```diff
@@ -581,7 +586,7 @@ static int tls1_check_cert_param(SSL *s, X509 *x, int set_ee_md)
      * Can't check curve_id for client certs as we don't have a supported
      * curves extension.
      */
-    rv = tls1_check_ec_key(s, s->server ? curve_id : NULL, &comp_id);
+    rv = tls1_check_ec_key(s, s->server ? curve_id : NULL, &comp_id, 0);
```

Comment:
I don't understand (for lack of familiarity with who calls this function, when and why, ...) why the last parameter is 0 here.  Perhaps a bit more commentary would make this more clear.

Reply:
Also, FWIW, in Postfix Wietse tends to avoid numeric boolean arguments, and always uses a `#define DONT_CHECK_OWN_GROUPS 0` and `#define CHECK_OWN_GROUPS 1` pattern to make the call sites more clear. Up to you whether you think this is a suitable pattern for OpenSSL.

Reply:
This is about checking that a certificate we are considering sending to the peer is suitable. This is actually the important change in this PR. Sending 0 here means we don't check that a certificate's group is in our own list of supported curves, i.e. if we've been configured with an ECDSA cert then we will use it if we can regardless. I've expanded out the comment about this call a bit. I've also added the suggested defines.